dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -232,7 +232,19 @@ else if ( !params.hasPeriods() && params.hasFilterPeriods() )
                 ProgramIndicator in = (ProgramIndicator) queryItem.getItem();
 
                 String asClause = " as " + quote( in.getUid() );
-                columns.add( "(" + programIndicatorService.getAnalyticsSql( in.getExpression(), in, params.getEarliestStartDate(), params.getLatestEndDate() ) + ")" + asClause );
+
+                String piSql = programIndicatorService.getAnalyticsSql( in.getExpression(), in, params.getEarliestStartDate(), params.getLatestEndDate() );
+
+                if ( queryItem.hasRelationshipType() )
```

Comment:
The strategy here of creating a subquery that triggers an aggregation seems solid. I think however that we need to add the program indicators filter ( in.getFilter() ) to the WHERE clause in the select below.